playbooks/files/rax-maas/plugins/ironic_capacity_check.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+
+# Copyright 2017, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import time
+
+import ipaddr
+from ironicclient import exc
+from maas_common import get_auth_ref
+from maas_common import get_ironic_client
+from maas_common import metric
+from maas_common import metric_bool
+from maas_common import print_output
+from maas_common import status_err
+from maas_common import status_ok
+
+
+def check(auth_ref, args):
+
+    IRONIC_ENDPOINT = ('http://{ip}:6385/v1'.format(ip=args.ip))
+
+    try:
+        if args.ip:
+            ironic = get_ironic_client(endpoint=IRONIC_ENDPOINT)
+        else:
+            ironic = get_ironic_client()
+
+        is_up = True
+
+    except exc.ClientException:
+        is_up = False
+    # Any other exception presumably isn't an API error
+    except Exception as e:
+        metric_bool('client_success', False, m_name='maas_ironic')
+        status_err(str(e), m_name='maas_ironic')
```

Comment:
Don't you need to return here? Otherwise, you'll call status_err and status_ok on L54.